### PR TITLE
20230306 schemas tablenames and packages

### PR DESCRIPTION
**Pull request by benjaminbollen**
superceding #67 (because I had a bunch of messy commits locally on that branch; restarted from latest develop)

on the good side:
```
all tables and columns found: [TableMetadata { table_name: "eth_block_headers", column_name: "at", column_type: "text" }, TableMetadata { table_name: "eth_block_headers", column_name: "hash", column_type: "text" }, TableMetadata { table_name: "eth_block_headers", column_name: "id", column_type: "text" }, TableMetadata { table_name: "eth_block_headers", column_name: "number", column_type: "integer" }, TableMetadata { table_name: "eth_block_headers", column_name: "parent_hash", column_type: "text" }, TableMetadata { table_name: "eth_block_headers", column_name: "timestamp", column_type: "text" }]
all tables found: {"eth_block_headers"}
```
these tables are returned

but then we still error on the later query:
```
thread 'main' panicked at 'Failed to load tables: DieselError(DatabaseError(Unknown, "schema \"$1\" does not exist"))', bin/node/cli/src/main.rs:111:29

postgres-ssp  | 2023-03-06 14:33:58.427 UTC [105] ERROR:  schema "$1" does not exist at character 221
postgres-ssp  | 2023-03-06 14:33:58.427 UTC [105] STATEMENT:  
postgres-ssp  |                     SELECT a.attname as pk
postgres-ssp  |                     FROM   pg_index i
postgres-ssp  |                     JOIN   pg_attribute a ON a.attrelid = i.indrelid
postgres-ssp  |                                         AND a.attnum = ANY(i.indkey)
postgres-ssp  |                     WHERE  i.indrelid = '$1.$2'::regclass
postgres-ssp  |                     AND    i.indisprimary;
postgres-ssp  |    
```

### CODE REVIEW

File: eureka-sink-postgres/db/src/db_loader.rs [hunk 1]
```diff
@@ -302,16 +307,21 @@ impl DBLoader {
...
+            WHERE  i.indrelid = '{}.{}'::regclass
             AND    i.indisprimary;
-        "
-        );
-
+            ",
+            self.schema.clone(), table);
+println!("query: {}", query);
+
+        // let primary_keys = sql_query(query)
```
Comment (benjaminbollen) [inline eureka-sink-postgres/db/src/db_loader.rs:317]: this should be cleaned up;

but if we don't flush we churn through the reponses, but simply accumulate them in memory

**Comment by benjaminbollen** [comment]
@jorgeantonio21 we are now failing on flush: primary key "id" is not part of the insert operation

```
thread 'main' panicked at 'Failed to execute query: FailedToExecuteQuery { query: "INSERT INTO ethereummainnet.eth_blockheaders (hash,number,parent_hash,timestamp) VALUES ('dfe2e70d6c116a541101cecbb256d7402d62125f6ddc9b607d49edc989825c64',100,'db10afd3efa45327eb284c83cc925bd9bd7966aea53067c1eebe0724d124ec1e','2015-07-30T15:34:03Z')", error: "null value in column \"id\" of relation \"eth_blockheaders\" violates not-null constraint" }', eureka-sink-postgres/db/src/flush.rs:31:30
```

File: bin/node/cli/src/main.rs [hunk 2]
```diff
@@ -196,9 +198,14 @@ async fn main() {
                         }
                         _ => {}
                     }
+                    // todo: flush is now per module output; it might make more sense per block?
+                    match db_loader.flush(output.name, cursor.clone()) {
```
Comment (jorgeantonio21) [inline bin/node/cli/src/main.rs:211]: This `output.name` would correspond to the primary key value of the `cursors` table. Does this make sense ?

**Review by vadim-frolov — APPROVED** [review]
we can merge this step and proceed with implementation and fixes

**Pull request merged**